packages/db-objects-redis/src/lib/objects/objectsInRedisClient.ts

```diff
@@ -1162,31 +1161,22 @@ export class ObjectsInRedisClient {
         );
     }
 
-    async _readFile(
-        id: string,
-        name: string,
-        options: CallOptions,
-        callback: ioBroker.ReadFileCallback,
-        meta: MetaObject
-    ) {
+    async _readFile(id: string, name: string, options: CallOptions, meta: MetaObject): ioBroker.ReadFilePromise {
         if (meta.notExists) {
-            return tools.maybeCallbackWithError(callback, ERRORS.ERROR_NOT_FOUND);
+            throw new Error(ERRORS.ERROR_NOT_FOUND);
         }
         if (!this.client) {
-            return tools.maybeCallbackWithError(callback, ERRORS.ERROR_DB_CLOSED);
+            throw new Error(ERRORS.ERROR_DB_CLOSED);
         }
 
         let buffer;
-        try {
-            buffer = await this._getBinaryState(this.getFileId(id, name, false));
-        } catch (e) {
-            return tools.maybeCallbackWithError(callback, e);
-        }
+        buffer = await this._getBinaryState(this.getFileId(id, name, false));
+
         const mimeType = meta && meta.mimeType;
         if (meta && !meta.binary && buffer) {
             buffer = buffer.toString();
         }
-        return tools.maybeCallbackWithError(callback, null, buffer, mimeType);
+        return { file: buffer, mimeType: mimeType as string };
```

Comment:
That's not correct. `mimeType` can be undefined. We should change the `ReadFilePromise` type if that is intended.

Reply:
Same goes for this line below
```
err ? reject(err) : resolve({ data: res, mimeType: mimeType as string })
```

Reply:
And the one in `readFileAsync`

Reply:
Haven't looked into ReadFile but I was naive and interpreted the typings

```typescript
type ReadFileCallback = (err?: NodeJS.ErrnoException | null, file?: Buffer | string, mimeType?: string) => void;
type ReadFilePromise = Promise<{ file: string | Buffer; mimeType: string }>;
        ```
         in a way that mimeType and file are only optional for Callback, because they will not be passed in error case, which would mean for Promise case, that they are always there on a resolved Promise.
         
But now looking code-wise if a `MetaObject` without `mimeType` is passed, it would be `undefined`. I will check the code, if we are currently ensuring in the other places that `mimeType` is defined, else we would need to adjust typings.

Reply:
Coming from `checkFile` it does not always exist.

Reply:
added it to https://github.com/ioBroker/adapter-core/issues/455 and will make expect errors instead of casting as string